include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -476,11 +494,18 @@ struct SessionOptionsImpl : Base<T> {
   SessionOptionsImpl& SetCustomCreateThreadFn(OrtCustomCreateThreadFn ort_custom_create_thread_fn);  ///< Wraps OrtApi::SessionOptionsSetCustomCreateThreadFn
   SessionOptionsImpl& SetCustomThreadCreationOptions(void* ort_custom_thread_creation_options);      ///< Wraps OrtApi::SessionOptionsSetCustomThreadCreationOptions
   SessionOptionsImpl& SetCustomJoinThreadFn(OrtCustomJoinThreadFn ort_custom_join_thread_fn);        ///< Wraps OrtApi::SessionOptionsSetCustomJoinThreadFn
+
+  void* RegisterCustomOpsLibrary(const char* library_path);  ///< Wraps OrtApi::RegisterCustomOpsLibrary
 };
 }  // namespace detail
 
-// No const version required
-using UnownedSessionOptions = detail::SessionOptionsImpl<detail::Unowned<OrtSessionOptions>>;
+using ConstSessionOptions = detail::ConstSessionOptionsImpl<detail::Unowned<const OrtSessionOptions>>;
+
+struct UnownedSessionOptions : detail::SessionOptionsImpl<detail::Unowned<OrtSessionOptions>> {
```

Comment:
>t UnownedSessionOptions :  [](http://example.com/codeflow?start=5&length=26)

Need a documentation for this and justifications since this deviates from a pattern.

Reply:
Reverted.